ims/einstein/iscsi/tgt.py

```diff
@@ -11,180 +12,154 @@
-        proc = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE,
-                                stderr=subprocess.PIPE)
-        output, err = proc.communicate(self.root_password +
-                                       "\n")
-        logger.debug("Output = %s, Error = %s", output, err)
-        if self.show_status() is 'Running':
+        """
+        Starts the tgt service
+
+        :return: None
+        """
+        try:
+            command = "service tgtd start"
+            output = shell.call(command, sudo=True)
+            logger.debug("Output = %s", output)
+            if self.show_status() is not 'Running':
+                raise iscsi_exceptions.StartFailedException()
+        except subprocess.CalledProcessError:
             raise iscsi_exceptions.StartFailedException()
 
     @log
     def stop_server(self):
-        '''
-        Need to check if this
-        :return:
-        '''
-        # arglist = self.arglist
-        # arglist.append("stop")
-        command = "service tgtd stop"
-        proc = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE,
-                                stderr=subprocess.PIPE)
-        output, err = proc.communicate(self.root_password + "\n")
-        logger.debug("Output = %s, Error = %s", output, err)
-        if self.show_status() is not 'Dead':
+        """
+        Stops the tgtd service
+
+        :return: None
+        """
+        try:
+            command = "service tgtd stop"
+            output = shell.call(command, sudo=True)
+            logger.debug("Output = %s", output)
+            if self.show_status() is not 'Dead':
+                raise iscsi_exceptions.StopFailedException()
+        except subprocess.CalledProcessError as e:
             raise iscsi_exceptions.StopFailedException()
 
     @log
     def restart_server(self):
-        '''
-        Again have to parse the output and send the values.
-        :return:
-        '''
+        """
+        Restarts the tgtd service
+
+        :return: None
+        """
         self.stop_server()
         self.start_server()
 
     @log
     def show_status(self):
-        # arglist = self.arglist
-        # arglist.append('status')
+        """
+        Returns the status of tgtd
+
+        :return: Running or Dead or Error as String
+        """
         command = "service tgtd status"
-        proc = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE,
-                                stderr=subprocess.PIPE)
-        status_string = proc.communicate(self.root_password + "\n")[0]
+        status_string = shell.call(command, sudo=True)
+        logger.debug("Output = %s", status_string)
         if 'active (running)' in status_string:
             return 'Running'
         elif 'inactive (dead)' in status_string:
             return 'Dead'
         # Have to check if there are any other states
         else:
-            return "Running in error state"
+            return "Error"
```

Comment:
It makes sense to raise an exception if start or stop of a service failed. This method is supposed to just return the status of the tgtd service, so I feel it doesn't make sense to raise an exception unless service tgtd status to execute. I agree it makes sense to add it in shell.py